cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/GcpMetadataHttpHandlerInternal.java

```diff
@@ -162,7 +172,7 @@ public void token(HttpRequest request, HttpResponder responder,
     try {
       // fetch token from credential provider
       ProvisionedCredential provisionedCredential =
-          credentialLoadingCache.get(
+          provisionedCredentialLoadingCache.get(
```

Comment:
Maybe we should change the response value of the cache to be `LoadingCache<ProvisionedCredentialCacheKey,
        GcpTokenResponse>`, since the `ProvisionedCredentialCacheKey` already contains a `scopes` field? Then we won't need two caches and instead can rely on a cached value prior to provisioning a credential.

Reply:
Merged both the caches.